src/libaktualizr/storage/invstorage.cc

```diff
@@ -174,56 +178,97 @@ std::shared_ptr<INvStorage> INvStorage::newStorage(const StorageConfig& config,
   }
 }
 
-void INvStorage::FSSToSQLS(FSStorageRead& fs_storage, SQLStorage& sql_storage) {
+bool INvStorage::FSSToSQLS(FSStorageRead& fs_storage, const std::shared_ptr<SQLStorage>& sql_storage,
+                           const bool check_exists) {
   std::string public_key;
   std::string private_key;
   if (fs_storage.loadPrimaryKeys(&public_key, &private_key)) {
-    sql_storage.storePrimaryKeys(public_key, private_key);
+    if (check_exists) {
```

Comment:
Maybe every 
```
    if (check_exists) {
      return true;
    } else {
      sql_storage->storeTlsCa(ca);
    }
```
change to: 

```
    if (!check_exists) {
      sql_storage->storeTlsCa(ca);
    } 
```
and `return true;` at the end of method

Reply:
The problem is that we want to migrate if any single file from FSStorage is found, but nothing is found, we shouldn't do anything with it. Currently, the migration is attempted if `storage.path` exists even if there is nothing in it. That is wrong and it causes an error with `aktualizr-info`.